src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/IEntitlementReader.cs

```diff
@@ -0,0 +1,27 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Common
+{
+    public interface IEntitlementReader
```

Comment:
Before introducing the interface, we had two classes (for reading CLI arguments and for reading tokens) with practically identical semantics and members for gathering all the `Errorable` properties required to generate a `NodeEntitlements` instance. But because they were separate types, the code for actually generating the `NodeEntitlements` was duplicated.

By extracting an interface out, the generation code (all the `With(...).Map(...)` stuff) can be written just once and tested independently (I put it in the `NodeEntitlements` class for now, and I suspect that could be simplified further, because the public `NodeEntitlements.WithXxx` methods are used (almost?) exclusively to support that generation code).

A second benefit is that when new entitlement properties are added (e.g. max-cores, batch account, etc.), they need to be added to the interface in order to support building the `NodeEntitlements` objects. That means they can't be inadvertently missed from the classes implementing those interfaces.